### PR TITLE
Don't link ft2font to zlib by default.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

First commit:

A default build of freetype (e.g. the one from MPLLOCALFREETYPE=1) uses
its own vendored version of zlib (see FT_CONFIG_OPTION_SYSTEM_ZLIB in
ftoption.h), so we don't need to link to zlib in that case; if we're
linking against a system freetype then we can assume that pkg-config
will list the right libraries to link.
https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/include/freetype/config/ftoption.h#n204

Not linking zlib explicitly means that the staticbuild option is not
used anymore and can just be stripped out.

Second commit:

Update build docs.
TL;DR: set MPLLOCALFREETYPE=1 and call it a day, unless you really know
what you want to do.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
This needs a re-base, but :+1:

**Comment by anntzer** [comment]
rebased.

**Comment by QuLogic** [comment]
I'm hesitant to merge while builds are failing.
